src/modules/monotype.js

```diff
@@ -54,30 +54,34 @@ goog.scope(function () {
     var self = this;
     var projectId = self.configuration_['projectId'];
     var version = self.configuration_['version'];
+    function checkAndLoadIfDownloaded() {
+	  if (loadWindow[Monotype.HOOK + projectId]) {
+	    var mti_fnts = loadWindow[Monotype.HOOK + projectId](),
+	    fonts = [];
 
+	    if (mti_fnts) {
+		  for (var i = 0; i < mti_fnts.length; i++) {
+		    fonts.push(new Font(mti_fnts[i]["fontfamily"]));
+		  }
+	    }
+	    onReady(fonts);
+	  } else {
+	    setTimeout(function () {
+	      checkAndLoadIfDownloaded();
+	    }, 0)
```

Comment:
In existing publish code, this information is available as soon as the script loaded. We need to pass all project fonts in which all fonts might not be using by the page. We're making change on our publish code such that it will send this information after parsing page. We're filtering unused project fonts and passing only those fonts which are being used on the page.